mllib/src/main/scala/org/apache/spark/ml/clustering/KMeans.scala

```diff
@@ -302,22 +302,19 @@ class KMeans @Since("1.5.0") (
 
   @Since("2.0.0")
   override def fit(dataset: Dataset[_]): KMeansModel = {
-    val handlePersistence = dataset.rdd.getStorageLevel == StorageLevel.NONE
-    fit(dataset, handlePersistence)
-  }
-
-  @Since("2.2.0")
-  protected def fit(dataset: Dataset[_], handlePersistence: Boolean): KMeansModel = {
```

Comment:
@hhbyyh, so that means that we should do the same things for others methods:
caching rdd before training for for:
-GaussianMixture.scala
-BisectingKMeans.scala
-LDA.scala
and remove the extrat train() method for all methods in the package classification.